src/desktop/secret.rs

```diff
@@ -93,11 +93,23 @@ impl<'a> std::ops::Deref for Secret<'a> {
 /// It crates a UnixStream internally for receiving the secret.
 pub async fn retrieve() -> Result<Vec<u8>, Error> {
     let proxy = Secret::new().await?;
+    let mut buf = Vec::with_capacity(64);
+
+    #[cfg(feature = "tokio")]
+    let mut x1 = {
+        let (x1, mut x2) = UnixStream::pair()?;
+        proxy.retrieve(&x2).await?;
+        x2.shutdown().await?;
+        x1
+    };
+    #[cfg(feature = "async-std")]
+    let mut x1 = {
+        let (x1, x2) = UnixStream::pair()?;
+        proxy.retrieve(&x2).await?;
+        x2.shutdown(Shutdown::Both)?;
```

Comment:
Afaik, there is no need to close both halves, closing the writing half should suffice.